File: myeong_da_yeon/week1/단속카메라.java
import java.util.*;

class Solution {
    public int solution(int[][] routes) {
        int answer = 1;
        
        Arrays.sort(routes, (o1,o2) -> {
            if(o1[0] == o2[0]) {
                return o1[1] - o2[1];
            } else {
                return o1[0] - o2[0];
            }
        });
        
        int tmp = routes[0][1];
        
        for(int i = 0;i<routes.length-1;i++) {
            if(tmp > routes[i][1]) {
                tmp = routes[i][1];
            }
            if(routes[i+1][0] > tmp) {
                answer++;
                tmp = routes[i+1][1];
            }
        }
        
        return answer;
    }
}

File: shin_jisoo/Greedy/단속카메라.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> routes) {
    int answer = 1;

    sort(routes.begin(), routes.end());

    int standard = routes[0][1];

    for (auto a : routes) {
        if (standard >= a[0]) {
            standard = min(a[1], standard);
        }
        else {
            answer++;
            standard = a[1];
        }
    }

    return answer;
}